plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/TrinoToMySqlWriteChecker.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.mysql;
+
+import io.trino.spi.TrinoException;
+
+import java.time.Instant;
+
+import static io.trino.spi.StandardErrorCode.INVALID_ARGUMENTS;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class TrinoToMySqlWriteChecker<T>
+{
+    // The range for TIMESTAMP values is '1970-01-01 00:00:01.000000' to '2038-01-19 03:14:07.499999'
+    public static final TrinoToMySqlWriteChecker<Instant> TIMESTAMP = new TrinoToMySqlWriteChecker<>(
```

Comment:
Is the name 'TIMESTAMP' something that has been overlooked, or a thought through variable name?

Reply:
We also seem to be using guice modules. Wouldn't it be cleaner if we configured a binding 'toInstance' in guice and injected it, rather than rely on statically initialized fields?

Reply:
The name is thought through - see the point of usage - `TIMESTAMP.validate(value)`.

Note that this class is "derived" from https://github.com/trinodb/trino/blob/5d8824edf9101eab451128bbad53d8c6ce930ad4/plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/TrinoToClickHouseWriteChecker.java.

And this class should go away if/when the MySQL JDBC driver bug gets fixed.